include/arch/x86/arch/machine/fpu.h

```diff
@@ -59,8 +34,17 @@ static inline uint32_t xsave_features_low(void)
 }
 
 /* Store state in the FPU registers into memory. */
-static inline void saveFpuState(user_fpu_state_t *dest)
+static inline void saveFpuState(arch_tcb_t *arch_tcb)
 {
+    user_fpu_state_t *dest = &arch_tcb->tcbContext.fpuState;
+#ifdef CONFIG_VTX
+    struct vcpu *vcpu = arch_tcb->tcbVCPU;
+
+    if (vcpu->fpu_active) {
```

Comment:
I might be missing some x86 details that means that the threads being passed in here and in `loadFpuState` must have a vcpu, but don't you need to check that the vcpu exists first?
```suggestion
    struct vcpu *vcpu = arch_tcb->tcbVCPU;

    if (vcpu && vcpu->fpu_active) {
```

Reply:
Indeed, not sure how I missed that.